azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py

```diff
@@ -51,14 +51,19 @@ def _execute_op(self, op):
             else:
                 client_id = op.device_id
 
+            # For MQTT, the entire user agent string should be appended to the username field in the connect packet
+            # For example, the username may look like this without custom parameters:
+            # yosephsandboxhub.azure-devices.net/alpha/?api-version=2018-06-30&DeviceClientType=py-azure-iot-device%2F2.0.0-preview.12
+            # The customer user agent string would simply be appended to the end of this username, in URL Encoded format.
```

Comment:
>in URL Encoded format. [](start = 99, length = 22)

Maybe add the note about URL-encoding to the user doc for this parameter.

Reply:
I've added documentation in the config.py stage, and in abstract_clients.py

---
In reply to: [338261318](https://github.com/Azure/azure-iot-sdk-python/pull/368#discussion_r338261318) [](ancestors = 338261318)